File: src/main/java/my/maryvk/maryvkweb/repository/RelationRepository.java
package my.maryvk.maryvkweb.repository;

import my.maryvk.maryvkweb.domain.Relation;
import my.maryvk.maryvkweb.domain.RelationType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

public interface RelationRepository extends JpaRepository<Relation, Long> {
    List<Relation> findAllByOwnerIdAndRelationTypeOrderByTargetId(Integer ownerId, RelationType relationType);
    @Transactional
    void deleteByOwnerIdAndTargetIdAndRelationType(Integer ownerId, Integer targetId, RelationType relationType);
}

File: src/main/java/my/maryvk/maryvkweb/repository/RelationRepository.kt
package my.maryvk.maryvkweb.repository

import my.maryvk.maryvkweb.domain.Relation
import my.maryvk.maryvkweb.domain.RelationType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.transaction.annotation.Transactional

interface RelationRepository : JpaRepository<Relation, Long> {
    fun findAllByOwnerIdAndRelationTypeOrderByTargetId(ownerId: Int?, relationType: RelationType): List<Relation>
    @Transactional
    fun deleteByOwnerIdAndTargetIdAndRelationType(ownerId: Int?, targetId: Int?, relationType: RelationType)
}